src/store/root-state.epics.ts

```diff
@@ -22,7 +24,11 @@ const rootStateResetEpic = (action$: Observable<Action>, state$: Observable<Root
     switchMap(keychainOptionsArray =>
       from(keychainOptionsArray).pipe(switchMap(options => Keychain.resetGenericPassword(options)))
     ),
-    mapTo(rootStateResetAction.success())
+    switchMap(() => from(BeaconHandler.getPermissions())),
+    switchMap(permissions =>
+      from(Promise.all(permissions.map(permission => BeaconHandler.removePermission(permission.accountIdentifier))))
```

Comment:
use `forkJoin` instead of `Promise.all`

Reply:
This process is about to be refactored, so there will be no need in it